src/Homeworks/Homework4/Queue.py

```diff
@@ -0,0 +1,42 @@
+from dataclasses import dataclass
+
+
+@dataclass
+class Queue:
+    size: int = 0
+    head: dict = None
+    tail: dict = None
+
+
+def empty(queue: Queue) -> bool:
+    return queue.size == 0
+
+
+def size(queue: Queue) -> int:
+    return queue.size
+
+
+def top(queue: Queue) -> any:
+    if not empty(queue):
+        return queue.head["value"]
+
+
+def push(queue: Queue, value: any) -> any:
+    new_element = {"value": value, "next": None}
+    if empty(queue):
+        queue.head = new_element
+        queue.tail = new_element
+    else:
+        queue.tail["next"] = new_element
+        queue.tail = new_element
+    queue.size += 1
+
+
+def pop(queue: Queue) -> any:
```

Comment:
Вкусно типы пописал. any это функция из стандартной библиотеки. И зачем указывать эни, если ничего не указать == вернуть Any? 

Вот тебе листинг mypy
```
Queue.py:7: error: Incompatible types in assignment (expression has type "None", variable has type "dict[Any, Any]")  [assignment]
Queue.py:8: error: Incompatible types in assignment (expression has type "None", variable has type "dict[Any, Any]")  [assignment]
Queue.py:19: error: Missing return statement  [return]
Queue.py:19: error: Function "builtins.any" is not valid as a type  [valid-type]
Queue.py:19: note: Perhaps you meant "typing.Any" instead of "any"?
Queue.py:24: error: Missing return statement  [return]
Queue.py:24: error: Function "builtins.any" is not valid as a type  [valid-type]
Queue.py:24: note: Perhaps you meant "typing.Any" instead of "any"?
Queue.py:35: error: Missing return statement  [return]
Queue.py:35: error: Function "builtins.any" is not valid as a type  [valid-type]
Queue.py:35: note: Perhaps you meant "typing.Any" instead of "any"?
Queue.py:41: error: Function "builtins.any" is not valid as a type  [valid-type]
Queue.py:41: note: Perhaps you meant "typing.Any" instead of "any"?
Found 9 errors in 1 file (checked 2 source files)
```